src/engine/input/InputManager.cs

```diff
@@ -0,0 +1,327 @@
+        // We reset our held down keys if the player tabs out while pressing a key
+        if (focus == MainLoop.NotificationWmFocusOut)
+        {
+            OnFocusLost();
+        }
+    }
+
+    internal static bool CallMethod(InputAttribute attribute, object[] parameters)
+    {
+        var method = attribute.Method;
+
+        // Do nothing if no method is associated
+        // TODO: it would be safer against mistakes to make it so that only specific attributes can have null method
+        if (method == null)
+            return true;
+
+        var disposed = new List<WeakReference>();
+        var instances = staticInstance.attributes[attribute];
+        var result = false;
+
+        if (method.IsStatic)
+        {
+            // Call the method without an instance if it's static
+            var invokeResult = method.Invoke(null, parameters);
+
+            if (invokeResult is bool asBool)
+            {
+                result = asBool;
+            }
+            else
+            {
+                result = true;
+            }
+        }
+        else
+        {
+            // Call the method for each instance
+            foreach (var instance in instances)
+            {
+                if (!instance.IsAlive)
+                {
+                    // if the WeakReference is no longer valid
+                    disposed.Add(instance);
+                    continue;
+                }
+
+                bool thisInstanceResult;
+                object invokeResult;
+
+                try
+                {
+                    invokeResult = method.Invoke(instance.Target, parameters);
+                }
+                catch (TargetInvocationException e)
+                {
+                    if (e.InnerException is ObjectDisposedException)
+                    {
+                        GD.PrintErr("A disposed object is still registered for input: ", e.InnerException);
+                    }
+                    else
+                    {
+                        GD.PrintErr("Failed to perform input method invoke: ", e);
+                    }
+
+                    disposed.Add(instance);
+                    continue;
+                }
+
+                if (invokeResult != null && invokeResult is bool asBool)
+                {
+                    thisInstanceResult = asBool;
+                }
+                else
+                {
+                    thisInstanceResult = true;
+                }
+
+                if (thisInstanceResult)
+                {
+                    result = true;
+                }
+            }
+        }
+
+        disposed.ForEach(p => instances.Remove(p));
+
+        return result;
+    }
+
+    private void OnInput(bool unhandledInput, InputEvent @event)
+    {
+        // Ignore mouse motion
+        // TODO: support mouse movement input as well
+        if (@event is InputEventMouseMotion)
+            return;
+
+        bool isUp = !@event.IsPressed();
```

Comment:
Wouldn't it make more sense to change this to isDown?

Reply:
I mean inverting the variable.
````suggestion
        bool isDown = @event.IsPressed();
```